cmip6_downscaling/analysis/analysis.py

```diff
@@ -0,0 +1,183 @@
+        elif var in ['pr']:
+            out_ds[var] = ds[var].resample(time='1MS').sum(dim='time')
+        else:
+            print(f'{var} not implemented')
+
+    return out_ds
+
+
+def annual_summary(ds):
+    """Creates an annual summary dataset.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Downscaled dataset
+    """
+    out_ds = xr.Dataset()
+    for var in ds:
+        if var in ['tasmax', 'tasmin']:
+            out_ds[var] = ds[var].resample(time='YS').mean()
+        elif var in ['pr']:
+            out_ds[var] = ds[var].resample(time='YS').sum()
+        else:
+            print(f'{var} not implemented')
+
+    return out_ds
+
+
+@task(log_stdout=True, tags=['dask-resource:TASKSLOTS=1'])
+def run_analyses(parameters, web_blob):
+    run_id = parameters['run_id']
+    workdir = '/home/jovyan/cmip6-downscaling/notebooks'
+    executed_notebook_path = f'{workdir}/analyses_{run_id}.ipynb'
+    executed_notebook_html_path = f'{workdir}/analyses_{run_id}.html'
+    pm.execute_notebook(
+        f'{workdir}/analyses.ipynb', executed_notebook_path, parameters=parameters
+    )  # TODO do these local paths work regardless of execution environment?
+
+    # convert from ipynb to html
+    os.system(f"jupyter nbconvert {executed_notebook_path} --to html")
+
+    blob_service_client = BlobServiceClient.from_connection_string(connection_string)
+    blob_name = web_blob + f'/analyses_{run_id}.html'
+    blob_client = blob_service_client.get_blob_client(container='$web', blob=blob_name)
+    # clean up before writing
+    try:
+        blob_client.delete_blob()
+    except:
+        pass
+
+    #  need to specify html content type so that it will render and not download
+    with open(executed_notebook_html_path, "rb") as data:
+        blob_client.upload_blob(data, content_settings=ContentSettings(content_type='text/html'))
+
+    return executed_notebook_path
+
+
+def load_top_cities(plot=False):
+    cities = pd.read_csv('/home/jovyan/cmip6-downscaling/notebooks/worldcities.csv')
+    top_cities = (
+        cities.sort_values('population', ascending=False)
+        .groupby('country')
+        .first()
+        .sort_values('population', ascending=False)[0:100][['city', 'lat', 'lng']]
+    )
+    additional_cities = [
+        'Seattle',
+        'Los Angeles',
+        'Denver',
+        'Chicago',
+        'Anchorage',
+        'Perth',
+        'Paramaribo',
+        'Fortaleza',
+    ]
+    for additional_city in additional_cities:
+        top_cities = top_cities.append(
+            cities[cities['city'] == additional_city][['city', 'lat', 'lng']]
+        )
+    if plot:
+        ax = plt.axes(projection=ccrs.PlateCarree())
+        ax.stock_img()
+        for (lat, lon) in top_cities[['lat', 'lng']].values:
+            plt.plot(
+                lon,
+                lat,
+                color='blue',
+                marker='o',
+            )
+    return top_cities
+
+
+def select_points(ds, top_cities):
```

Comment:
```suggestion
def select_points(ds: xr.Dataset, top_cities: pd.DataFrame) -> xr.Dataset:
```

+ add a docstring

Reply:
so many docstrings and typing things! my fingers are tired from all their tapping.